simple/stats/data.py

```diff
@@ -234,25 +235,45 @@ class Observation:
   date: str
   value: str
   provenance: str
+  unit: str = ""
+  scaling_factor: str = ""
+  measurement_method: str = ""
+  observation_period: str = ""
   properties: dict[str, str] = field(default_factory=dict)
 
   def __post_init__(self):
     if not self.properties:
       self.properties = {}
-    # Properties in the DB are stored as gzipped and base64 encoded strings.
-    # Convert it to json / dict so it is available as a dict in code.
+    # Properties in the DB are stored as stringified json.
+    # Load it as a dict in code.
     elif isinstance(self.properties, str):
-      self.properties = base64_decode_and_gunzip_json(self.properties)
-
-  def properties_string(self) -> str:
-    if not self.properties:
-      return ""
-    return gzip_and_base64_encode_json(self.properties)
+      self.properties = json.loads(self.properties)
+
+    self.unit = self.unit or self.properties.get(sc.PREDICATE_UNIT, "")
+    self.scaling_factor = self.scaling_factor or self.properties.get(
+        sc.PREDICATE_SCALING_FACTOR, "")
+    self.measurement_method = self.measurement_method or self.properties.get(
+        sc.PREDICATE_MEASUREMENT_METHOD, "")
+    self.observation_period = self.observation_period or self.properties.get(
+        sc.PREDICATE_OBSERVATION_PERIOD, "")
```

Comment:
+1 the properties should not have the 4 special props

Reply:
Simplified this class and moved properties extraction to a dedicated `ObservationProperties` class. Note that the latter is still needed since users will specify all properties in a single json object in `config.json` and the standard properties need to be parsed out.

I agree it should not be in `Observation` itself so separated it out.